pkg/apis/cyclone/v1alpha1/workflow.go

```diff
@@ -0,0 +1,30 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Workflow defines a workflow
+type Workflow struct {
+	// Metadata for the resource, like kind and apiversion
+	metav1.TypeMeta `json:",inline"`
+	// Metadata for the particular object, including name, namespace, labels, etc
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Workflow specification
+	Spec WorkflowSpec `json:"spec"`
+}
+
+// WorkflowSpec defines workflow specification.
+type WorkflowSpec struct {
+	Stages StageItem `json:"stageItem"`
+}
+
+// StageItem describes a stage in a workflow.
+type StageItem struct {
+	// Name of stage
+	Name string `json:"name"`
+	// Stages that this stage depends on
+	Depends []string `json:"depends"`
```

Comment:
FYI: a lot of places use stage name as a reference to a stage, but name is really not a unique identifier (e.g. name can change; name can be the same across different namespace).

I think we can stand with this now...